Api/Data/NodeInterface.php

```diff
@@ -3,5 +3,215 @@
+    const POSITION = 'position';
+    const LEVEL = 'level';
+    const TITLE = 'title';
+    const TARGET = 'target';
+    const CREATION_TIME = 'creation_time';
+    const UPDATE_TIME = 'update_time';
+    const IS_ACTIVE = 'is_active';
 
-}
\ No newline at end of file
+    /**
+     * Get node id
+     *
+     * @return int
+     */
+    public function getNodeId();
+
+    /**
+     * Set node id
+     *
+     * @param int $nodeId
+     * @return $this
+     */
+    public function setNodeId($nodeId);
+
+    /**
+     * Get menu id
+     *
+     * @return int
+     */
+    public function getMenuId();
+
+    /**
+     * Set menu id
+     *
+     * @param int $menuId
+     * @return $this
+     */
+    public function setMenuId($menuId);
+
+    /**
+     * Get type
+     *
+     * @return string
+     */
+    public function getType();
+
+    /**
+     * Set type
+     *
+     * @param string $type
+     * @return $this
+     */
+    public function setType($type);
+
+    /**
+     * Get content
+     *
+     * @return string
+     */
+    public function getContent();
+
+    /**
+     * Set content
+     *
+     * @param string $content
+     * @return $this
+     */
+    public function setContent($content);
+
+    /**
+     * Get classes
+     *
+     * @return string
+     */
+    public function getClasses();
+
+    /**
+     * Set classes
+     *
+     * @param string $classes
+     * @return $this
+     */
+    public function setClasses($classes);
+
+    /**
+     * Get parent id
+     *
+     * @return int
+     */
+    public function getParentId();
+
+    /**
+     * Set parent id
+     *
+     * @param int $parentId
+     * @return $this
+     */
+    public function setParentId($parentId);
+
+    /**
+     * Get position
+     *
+     * @return int
+     */
+    public function getPosition();
+
+    /**
+     * Set position
+     *
+     * @param int $position
+     * @return $this
+     */
+    public function setPosition($position);
+
+    /**
+     * Get level
+     *
+     * @return int
+     */
+    public function getLevel();
+
+    /**
+     * Set level
+     *
+     * @param int $level
+     * @return $this
+     */
+    public function setLevel($level);
+
+    /**
+     * Get title
+     *
+     * @return string
+     */
+    public function getTitle();
+
+    /**
+     * Set title
+     *
+     * @param string $title
+     * @return $this
+     */
+    public function setTitle($title);
+
+    /**
+     * Get target
+     *
+     * @return int
+     */
+    public function getTarget();
+
+    /**
+     * Set target
+     *
+     * @param int $target
+     * @return $this
+     */
+    public function setTarget($target);
+
+    /**
+     * Get creation time
+     *
+     * @return string
+     */
+    public function getCreationTime();
+
+    /**
+     * Set creation time
+     *
+     * @param string $creationTime
+     * @return $this
+     */
+    public function setCreationTime($creationTime);
```

Comment:
Does Magento cann't properly handle real `DateTime`? Plain strings looks a little bit wierd. Even more - we don't know what format to use here,

Reply:
Magento can not handle `DATETIME` so I left the type and added the format